_posts/2022-04-21-ambient-peer-discovery.md

```diff
@@ -0,0 +1,334 @@
+offering high robustness and resilience against attacks.
+Gossipsub implements many improvements overcoming the shortcomings typically associated with unstructured P2P networks, e.g. inefficient flooding based routing.
+The gossipsub mesh network is managed in a decentralized way, which requires each node to know other participating peers.
+Waku v2 may use any combination of its ambient discovery protocols to find appropriate peers.
+
+Summarizing, Waku v2 comprises a *peer management layer* based on libp2p gossipsub,
+which manages the peers of nodes, and an *ambient peer discovery layer*,
+which provides information about peers to the peer management layer.
+
+We focus on ambient peer discovery methods that are in line with our goal of building a fully decentralized, generalized, privacy-preserving and censorship-resistant messaging protocol.
+Some of these protocols still need adjustments to adhere to our privacy and anonymity requirements. For now, we focus on operational stability and feasibility.
+However, when choosing techniques, we pay attention to selecting mechanisms that can feasibly be tweaked for privacy in future research efforts.
+Because of the modular design and the fact that Waku v2 has several discovery methods at its disposal, we could even remove a protocol in case future evaluation deems it not fitting our standards.
+
+This post covers the current state and future considerations of ambient peer discovery for Waku v2,
+and gives reason for changes and modifications we made or plan to make.
+The ambient peer discovery protocols currently supported by Waku v2 are a modified version of Ethereum's [Discovery v5](https://github.com/ethereum/devp2p/blob/6b0abc3d956a626c28dce1307ee9f546db17b6bd/discv5/discv5.md)
+and [DNS-based discovery](https://vac.dev/dns-based-discovery).
+Waku v2 further supports [gossipsub's peer exchange protocol](https://github.com/libp2p/specs/blob/10712c55ab309086a52eec7d25f294df4fa96528/pubsub/gossipsub/gossipsub-v1.1.md#prune-backoff-and-peer-exchange).
+In addition, we plan to introduce protocols for general peer exchange and capability discovery, respectively.
+The former allows resource restricted nodes to outsource querying for peers to stronger peers,
+the latter allows querying peers for their supported capabilities.
+Besides these new protocols, we are working on integrating capability discovery in our existing ambient peer discovery protocols.
+
+## Static Node Lists
+
+The simplest method of learning about peers in a P2P network is via static node lists.
+These can be given to nodes as start-up parameters or listed in a config-file.
+They can also be provided in a script-parseable format, e.g. the [status fleets](https://fleets.status.im/) in JSON.
+While this method of providing bootstrap nodes is very easy to implement, it requires static peers, which introduce centralized elements.
+Also, updating static peer information introduces administrative overhead.
```

Comment:
```suggestion
Also, updating static peer information introduces administrative overhead.
```
I'd be more specific and clarify that you have to change code and then release the code.
DNS-based Discovery also has administrative overhead as you have to update the DNS record.
However, with DNS Discovery you are able to separate code/release management and bootstrap node management.